src/w3d/renderer/dx8wrapper.h

```diff
@@ -714,6 +716,21 @@ inline unsigned int DX8Wrapper::Convert_Color(const Vector4 &color)
     return D3DCOLOR_COLORVALUE(color.X, color.Y, color.Z, color.W);
 }
 
+inline void DX8Wrapper::Clamp_Color(Vector4 &color)
+{
+    color.X = std::clamp(color.X, 0.f, 1.f);
+    color.Y = std::clamp(color.Y, 0.f, 1.f);
+    color.Z = std::clamp(color.Z, 0.f, 1.f);
+    color.W = std::clamp(color.W, 0.f, 1.f);
+}
+
+inline unsigned int DX8Wrapper::Convert_Color_Clamp(const Vector4 &color)
+{
+    Vector4 clamped_color = color;
+    DX8Wrapper::Clamp_Color(clamped_color);
+    return Convert_Color(reinterpret_cast<const Vector3 &>(clamped_color), clamped_color[3]);
```

Comment:
You can just pass it the vector4 there is already an overload for that and avoids the UB of reinterpret cast to convert a vec4 to a vec3.